docs/guide.rst

```diff
@@ -31,109 +31,116 @@ The `WGPU <https://github.com/pygfx/wgpu-py>`_ library provides the low level AP
+worlds/scenarios to render) using three main ingredients: (1) `objects` and
+their visual properties, (2) `light` sources, and (3) a `camera` that sees
+things. Once we defined those three things, we can position them in our scene
+and then use a `renderer` to look at what we have created (or take
+pictures/videos of it).
 
-.. code-block:: python
-
-    scene = gfx.Scene()
-
-    geometry = gfx.BoxGeometry(200, 200, 200)
-    material = gfx.MeshPhongMaterial(color=(1, 1, 0, 01))
-    cube = gfx.Mesh(geometry, material)
-
-    scene.add(cube)
-
-
-Opening a window to render to
------------------------------
-
-Your visualization must end up on the screen somehow. For this, we use the
-canvas abstraction provided by wgpu-py.
+.. note:: 
+    The :ref:`full code example <full_example>` can be found below.
 
-.. code-block:: python
+Let's look at a hello world example of how this works: A rotating cube. We begin
+by defining an empty `Scene`::
 
-    # Create Qt widget that can function as a canvas
-    from wgpu.gui.qt import WgpuCanvas
-    canvas = WgpuCanvas()
+    import pygfx as gfx
 
+    scene = gfx.Scene()
 
-We can ask the canvas to schedule a draw event, and tell it what to do
-to perform a draw.
+Right now this scene is very desolate. There is no light, no objects, and
+nothing that can look at those objects. Let's change this by adding some
+`light` and a `camera`::
 
-.. code-block:: python
+    # and god said ...
+    scene.add(gfx.AmbientLight())
 
-    def animate():
-       ...  # we'll get to this
+    camera = gfx.PerspectiveCamera(70, 16 / 9)
+    camera.position.z = 400
 
+Now there is light and a camera to perceive the light. To complete the setup
+we also need to add an object to look at::
 
-    canvas.request_draw(animate)
+    geometry = gfx.box_geometry(200, 200, 200)
+    material = gfx.MeshPhongMaterial(color=(1, 1, 0, 1))
+    cube = gfx.Mesh(geometry, material)
 
+    scene.add(cube)
 
-Setting up a renderer
----------------------
+Objects are slightly more complicated than lights or cameras. They have a
+`geometry`, which controlls an object's form, and a `material`, which controls
+an object's appearance (color, reflectiveness, etc).
 
-To render your scene to the canvas, you need a renderer. A renderer
-needs a target to render to (which can be a canvas or a texture).
-And finally, you need a camera to define the point of view to render the scene from.
+Now we have all the necessary ingredients and it is time to take a look. To do
+so we need to create a `canvas` to draw what we see (here an on-screen window)
+and a `renderer` that will look at the scene (through the camera) and draw what
+it sees onto the canvas::
 
-.. code-block:: python
+    from wgpu.gui.auto import WgpuCanvas, run
 
-    # A renderer is associated with a canvas (or a texture) that it renders to
-    renderer = gfx.renderers.WgpuRenderer(canvas)
+    renderer = gfx.renderers.WgpuRenderer(WgpuCanvas())
+    renderer.render(scene, camera)
```

Comment:
The `renderer.render()` call should be done in the "draw event".
```suggestion
    renderer.request_draw(lambda: renderer.render(scene, camera))
```

Lambda's are a bit ugly. Maybe it makes sense to create an `animate()` function here with just that one line, and explain the idea. Then down below the animate function is extended.

Reply:
oh? Did I get that right that you aren't supposed to call `renderer.render` from outside a `draw_function`?

Reply:
Sorry for being a bit vague about this. I had to dig into the code a bit to get it straight :)  Let me try to write a piece of text that could be put in the guide (feel free to edit it to your style).

The `renderer.render(scene, camera)` call renders the scene, from the viewpoint of the camera, to the internal texture of the canvas. This does not put the image onto the screen yet; we need to request the canvas to perform a draw using `renderer.request_draw(animate)`. This makes the GUI schedule a draw event, in which it will call the given function, and then present the texture to the screen. Note that `renderer.request_draw()` can be called (without an argument) from hereon to schedule new draws.

Reply:
I've updated the guide to reflect this. @almarklein Please double check if I got the logic right and this makes sense.